spec/src/main/asciidoc/architecture.adoc

```diff
@@ -195,7 +195,7 @@ There is one `MetricRegistry` instance for each of the predefined scopes listed
 Metrics can be added to or retrieved from the registry either using the `@Metric` annotation
 (see <<app-programming-model#api-annotations, Metrics Annotations>>) or using the `MetricRegistry` object directly.
 
-A metric is uniquely identified by the `MetricRegistry` if the `MetricID` associated with the metric is unique. That is to say, there are no other metrics with the same combination of metric name and tags. However, all metrics of the same name must be of the same type otherwise an `IllegalArgumentException` will be thrown. This exception will be thrown during registration.
+A metric is uniquely identified by the `MetricRegistry` if the `MetricID` associated with the metric is unique. That is to say, there are no other metrics with the same combination of metric name and tags. However, all metrics of the same name must be of the same type and contain the same set of tag names otherwise an `IllegalArgumentException` will be thrown. This exception will be thrown during registration.
```

Comment:
1. Tiny nit - I don't think of a metric as "containing" the tags used to identify it. Maybe:
   
   "and contain" -> "and be identified by"
   
   or something similar.
2. In the "Reusing Metrics" section below... We mention the requirement for consistency of metadata there also. We should add a mention there of the requirement for consistency of the tag name set. Perhaps something like:
   
   "The prerequisite for this is that..." -> 
   
   "All such annotations must carry the same metadata and tag names. If multiple annotations declare the same metric but contain different metadata or tag names, an `IllegalArgumentException` must be thrown during startup."